components/windows/include/windows/context.hpp

```diff
@@ -0,0 +1,38 @@
+/* Copyright (c) 2023, Thomas Atkinson
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 the "License";
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <string>
+
+#include <Windows.h>
+#include <core/platform/context.hpp>
+
+namespace vkb
+{
+/**
+ * @brief MS Windows platform context
+ *
+ * @warning Use in extreme circumstances with code guarded by the PLATFORM__WINDOWS define
```

Comment:
Just curious, but what does "use in extreme circumstances..." mean?

Reply:
We don't want to pollute the code base with platform specific code, this is just a warning against that. I didn't spend too much time thinking about the comments as I am likely to do a more formal documentation pass later